### DISCUSSION TITLE
[License registration failed] `unknown license type OLD`

### DISCUSSION THREAD

**Discussion by murase-dev** [dashingsoft/pyarmor#1233] [Q&A] [answered]
I tried to register the license, but the following error occured.

pyarmor reg -p CS pyarmor-regcode-3684.txt
INFO     Python 3.7.9
INFO     Pyarmor 8.1.9 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     prepare "pyarmor-regcode-3684.txt"
INFO     query key file from server
ERROR    this license does not work in Pyarmor 8
ERROR    please check Pyarmor 8.0 EULA
ERROR    unknown license type OLD

Reffering to the following URL, I also tried to upgrade license, but the following error occured.
https://pyarmor.readthedocs.io/en/latest/licenses.html#upgrading-old-license

INFO     prepare "pyarmor-regcode-3684.txt"
INFO     query key file from server
ERROR    please check Pyarmor 8.0 EULA
ERROR    old code "None" can not be upgraded

**Comment by jondy** [comment]
Please check documentation to find the reason

**Comment by jondy** [comment]
How about upgrade to pyarmor 8.2.0, it's released just now

**Comment by murase-dev** [comment]
Thank you for your prompt response and introduction of the document.
https://github.com/dashingsoft/pyarmor/issues/980
But the problem is not solved.

> Do not upgrade to 8.0 if don't understand what're changed, and it need some time to handle upgrading issues.
→ First, I tried to register the licence of the pyarmor installed for the first time(not upgraded). 

> Upon the release of Pyarmor 8.0, there are three scenarios for existing users:
> 1. Never upgrade to version 8.0+
→ After installing pyarmor 7.7.4, I tried to register the licence of it, but the command written in "pyarmor-regcode-3684.txt" didn't work:
pyarmor reg -p XXX pyarmor-regcode-3684.txt

> 2. Upgrade to version 8.0 but only use old features
> 3. Upgrade to version 8.0 and use new features
→ Upgrading the licence failed, as mentioned above.

> How about upgrade to pyarmor 8.2.0, it's released just now
→ I tried it, but the error occured as above:
INFO     Python 3.7.9
INFO     Pyarmor 8.2.0 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     prepare "pyarmor-regcode-3684.txt"
INFO     query key file from server
ERROR    this license does not work in Pyarmor 8
ERROR    please check Pyarmor 8.0 EULA
ERROR    unknown license type OLD

**Comment by jondy** [comment]
Please use `-u` to upgrade your license in Pyarmor 8.2.0

**Reply by murase-dev** [reply]
Thank you, but the following error occured.

Are you sure to continue? (yes/no) yes
INFO     process activation file "pyarmor-regcode-3684.txt"
INFO     send request to server
INFO     write registration file "pyarmor-regfile-3684.zip"
INFO     register "pyarmor-regfile-3684.zip"
INFO     update license token
ERROR    request license token faild (104)

**Reply by jondy** [reply]
> ERROR request license token faild (104)

This error means connect to license server failed, please check your firmware log, make sure it doesn't block the request.

**Reply by murase-dev** [reply]
Sorry, I couldn't find any such logs.
And I temporarily turned off the firewall and tried again, but the situation didn't change.

**Reply by murase-dev** [reply]
We have already paid our fees but are unable to use pyarmor.
Could you please reissue the license file?

**Comment by jondy** [comment, marked as answer]
Hi guy, please install Pyarmor 7.7.4 and check Pyarmor 7.x documentation, I'm sure this license works for Pyarmor 7.7.4 if you use it in right way.

Do not check Pyarmor latest documentation, it's totally different from Pyarmor 7.x

**Reply by murase-dev** [reply]
It seemed to work file. Thank you very much for your advice.

pyarmor register pyarmor-regcode-3684.txt
INFO     PyArmor Version 7.7.4
INFO     Python 3.7.9
INFO     Read registration code file: pyarmor-regcode-3684.txt
INFO     Start to activate this code
INFO     Got keyfile of this code, this code is activated
INFO     Start to register keyfile: pyarmor-regfile-3684.zip
INFO     Save registration data to: C:\Users\XXXX\.pyarmor
INFO     Extracting license.lic
INFO     Extracting .pyarmor_capsule.zip
INFO     This keyfile has been registered successfully.
INFO     Run "pyarmor register" to check registration information.
